src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/DeleteUserProfileServiceImpl.java

```diff
@@ -0,0 +1,72 @@
+package uk.gov.hmcts.reform.userprofileapi.service.impl;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.reform.userprofileapi.controller.request.UserProfileDataRequest;
+import uk.gov.hmcts.reform.userprofileapi.controller.response.UserProfilesDeletionResponse;
+import uk.gov.hmcts.reform.userprofileapi.domain.entities.UserProfile;
+import uk.gov.hmcts.reform.userprofileapi.exception.ResourceNotFoundException;
+import uk.gov.hmcts.reform.userprofileapi.repository.UserProfileRepository;
+import uk.gov.hmcts.reform.userprofileapi.service.AuditService;
+import uk.gov.hmcts.reform.userprofileapi.service.DeleteResourceService;
+
+@Service
+@Slf4j
+public class DeleteUserProfileServiceImpl implements DeleteResourceService<UserProfileDataRequest> {
+
+    @Autowired
+    private UserProfileRepository userProfileRepository;
+
+    @Autowired
+    private AuditService auditService;
+
+    @Override
+    @Transactional
+    public UserProfilesDeletionResponse delete(UserProfileDataRequest profilesData) {
+
+        List<UserProfile> userProfiles = new ArrayList<UserProfile>();
+        Set<String> userIds = new HashSet<String>(profilesData.getUserIds());
+        userIds.forEach(userId -> userProfiles.add(validateUserStatus(userId)));
+
+        return deleteUserProfiles(userProfiles);
+    }
+
+    /**
+     * Either delete all the audit and userProfiles data from the data base or none.
+     *
```

Comment:
can u change the comment here?.. i dont think u r deleting anything from the response table

Reply:
I am deleting the entries from response and user profile table.